src/composables/transactionData.ts

```diff
@@ -180,30 +180,44 @@ export function useTransactionData({
       return [];
     }
 
+    let convertToCoin = false;
+
     const { protocol = PROTOCOLS.aeternity } = activeTransaction.value;
     const adapter = ProtocolAdapterFactory.getAdapter(protocol);
     const protocolTokens = getProtocolAvailableTokens(protocol);
 
-    // AE DEX and wrapped AE (WAE)
-    // TODO move this logic to adapter and store resolved data in the transactions
-    if (
-      innerTx.value?.function
-      && (!isDexAllowance.value || showDetailedAllowanceInfo)
-    ) {
-      const functionName = camelCase(innerTx.value?.function) as TxFunctionParsed;
-      const functionResolver = getTransactionTokenInfoResolver(functionName);
-
-      if (functionResolver) {
-        return functionResolver({ tx: outerTx.value } as ITransaction, protocolTokens)
-          .tokens
-          .map(({
-            amount,
-            decimals,
-            ...otherAssetData
-          }) => ({
-            amount: +toShiftedBigNumber(amount!, -decimals!),
-            ...otherAssetData,
-          }));
+    // TODO move AE specific logic to adapter and store resolved data in the transactions
+    if (protocol === PROTOCOLS.aeternity) {
+      // AE DEX and wrapped AE (WAE)
+      if (
+        innerTx.value?.function
+        && (!isDexAllowance.value || showDetailedAllowanceInfo)
+      ) {
+        const functionName = camelCase(innerTx.value.function) as TxFunctionParsed;
+        const functionResolver = getTransactionTokenInfoResolver(functionName);
+
+        if (functionResolver) {
+          return functionResolver({ tx: outerTx.value } as ITransaction, protocolTokens)
+            .tokens
+            .map(({
+              amount,
+              decimals,
+              ...otherAssetData
+            }) => ({
+              amount: +toShiftedBigNumber(amount!, -decimals!),
+              ...otherAssetData,
+            }));
+        }
+
+        // Convert all unresolvable transactions that are not transfers as a coin.
+        if (functionName !== TX_FUNCTIONS.transfer) {
```

Comment:
This is valid only if the `contractId` in the transaction is from known `protocolTokens`.